src/Integration.Vsix.UnitTests/Analysis/IssueHandlerTests.cs

```diff
@@ -177,37 +168,38 @@ public void HandleNewIssues_HasFileLevelIssue_IssueIsReturned()
 
             // Assert
             notificationHandler.InvocationCount.Should().Be(1);
-            notificationHandler.UpdatedSnapshot.Issues.Count().Should().Be(1);
+            notificationHandler.UpdatedSnapshot.Issues.Count().Should().Be(0);
         }
 
         [TestMethod]
-        public void HandleNewIssues_NoIssues_NoIssuesReturned()
+        public void HandleNewIssues_SomeSuppressedIssues_IssuesGetMarkedCorrectly()
         {
             // Arrange
-            var inputIssues = Enumerable.Empty<IAnalysisIssueVisualization>();
+            var issue1 = CreateIssue("xxx", startLine: 1, endLine: 1);
+            var issue2 = CreateIssue("xxx2", startLine: 2, endLine: 2);
+            var issue3 = CreateIssue("xxx3", startLine: 3, endLine: 3);
+            var issue4 = CreateIssue("xxx4", startLine: 4, endLine: 4);
+
+            var issues = new[] { issue1, issue2, issue3, issue4 };
+            var issuesFilter = new Mock<IIssuesFilter>();
+            issuesFilter.Setup(x => x.GetMatches(issues)).Returns(new[] { issue1, issue4 });
 
             var notificationHandler = new SnapshotChangeHandler();
-            var testSubject = CreateTestSubject(notificationHandler.OnSnapshotChanged);
+            var testSubject = CreateTestSubject(notificationHandler.OnSnapshotChanged, issuesFilter: issuesFilter.Object);
 
             // Act
-            testSubject.HandleNewIssues(inputIssues);
+            testSubject.HandleNewIssues(issues);
 
             // Assert
             notificationHandler.InvocationCount.Should().Be(1);
-            notificationHandler.UpdatedSnapshot.Issues.Count().Should().Be(0);
-        }
+            notificationHandler.UpdatedSnapshot.Issues.Count().Should().Be(4);
 
-        private static IIssuesFilter CreatePassthroughIssuesFilter()
-            => CreateIssuesFilter(null).Object;
+            issue1.IsSuppressed = true;
 
-        private static Mock<IIssuesFilter> CreateIssuesFilter(IEnumerable<IFilterableIssue> optionalDataToReturn = null)
```

Comment:
This is not really needed anymore as calling the issue filter does not alter the list of issues anymore.